### PR TITLE
Add typos - Source code spell checker 

### PR DESCRIPTION
**Pull request by ameknite**
Same as https://github.com/rust-windowing/winit/pull/3557 but with [typos](https://github.com/crate-ci/typos)

Advantages:

 -  Fast to run on monorepos
 -  Low false positives so you can run on PRs

For reference typos is used in:
Bevy - https://github.com/bevyengine/bevy/pull/12036
Wgpu - https://github.com/gfx-rs/wgpu/pull/5191

### CODE REVIEW

**Comment by notgull** [comment]
Looks like the current implementation takes around 8 seconds. So it's faster and it seems more thorough as well. So I'm ready to switch to this instead of `cspell`.

@rust-windowing/winit Thoughts?

File: .github/workflows/ci.yml
Comment (daxpedda) [inline .github/workflows/ci.yml:27]: I would prefer if we stick to the known [`taiki-e/install-action`](https://github.com/taiki-e/install-action) instead.

File: .github/workflows/ci.yml
Comment (daxpedda) [inline .github/workflows/ci.yml:35]: Would prefer if we don't recommend extensions like this unless we know them well.

**Review by daxpedda — CHANGES_REQUESTED** [review]
I've used `typos` a lot by now and I had a great experience!
I don't have anything against `cspell` and I also don't mind running multiple spellcheckers as well.

Except the couple of nits, LGTM.

File: .github/workflows/ci.yml
Comment (kchibisov) [reply .github/workflows/ci.yml:35]: Yeah, I'd suggest to drop all of that.

**Comment by ameknite** [comment]
This fixes the same typos as #3556. Do I removed them from here and wait for the other pr to be merged, or I keep them?

**Comment by kchibisov** [comment]
>This fixes the same typos as https://github.com/rust-windowing/winit/pull/3556. Do I removed them from here and wait for the other pr to be merged, or I keep them?

No, will merge your patch instead fixing them.

**Review by kchibisov — APPROVED** [review]
Could you squash your changes?

**Comment by ameknite** [comment]
squashed

File: typos.toml
Comment (daxpedda) [inline typos.toml:1]: Instead of leaving all these placeholder tables and comments, a link to the documentation of this configuration file would be better I believe.

**Review by daxpedda — APPROVED** [review]
LGTM except the small nit.

File: typos.toml
Comment (ameknite) [reply typos.toml:1]: ok

File: .github/workflows/ci.yml
Comment (daxpedda) [reply .github/workflows/ci.yml]: Ha, I didn't realize that the `crate-ci/typos` action is the same repo as the `typos` project itself.
Well, its fine either way.

**Review by daxpedda — APPROVED** [review]

**Review by notgull — APPROVED** [review]
Looks good to me. Once this is rebased I will merge.

**Comment by kchibisov** [comment]
@ameknite I've rebased things myself.

**Pull request merged**
